fmriprep/workflows/bold.py

```diff
@@ -220,12 +221,18 @@ def init_func_preproc_wf(bold_file, ignore, freesurfer,
             'magnitude1': 'sub-03/ses-2/fmap/sub-03_ses-2_run-1_magnitude1.nii.gz',
             'magnitude2': 'sub-03/ses-2/fmap/sub-03_ses-2_run-1_magnitude2.nii.gz'
         }]
+        run_stc = True
     else:
         metadata = layout.get_metadata(bold_file)
         # Find fieldmaps. Options: (phase1|phase2|phasediff|epi|fieldmap)
         fmaps = layout.get_fieldmap(bold_file, return_list=True) \
             if 'fieldmaps' not in ignore else []
 
+        if "SliceTiming" in metadata and 'slicetiming' not in ignore:
```

Comment:
Please consider taking out this switch, so that `run_stc` is always set to a sensible value:

Insert in L211:
```
# Check if slice timing correction is to be executed
run_stc = "SliceTiming" in metadata and 'slicetiming' not in ignore
```

Then current line 224 stays unchanged. And this block (L231-234) is reduced to one line:
```
run_stc = run_stc and (_get_series_len(bold_file) > 4 or 'TooShort')
```

Reply:
`metadata` is defined after L211. But I can simplify these lines.